pkg/app/mutate_app_test.go

```diff
@@ -318,6 +318,82 @@ func Test_MutateApp(t *testing.T) {
 			operation:       admissionv1.Create,
 			expectedPatches: nil,
 		},
+		{
+			name:   "case 8: flawless flow for org-namespaced app",
+			oldObj: v1alpha1.App{},
+			obj: v1alpha1.App{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "kiam",
+					Namespace: "org-eggs2",
+					Labels: map[string]string{
+						label.Cluster: "eggs2",
+					},
+				},
+				Spec: v1alpha1.AppSpec{
+					Catalog:   "giantswarm",
+					Name:      "kiam",
+					Namespace: "kube-system",
+					KubeConfig: v1alpha1.AppSpecKubeConfig{
+						InCluster: false,
+					},
+					Version: "1.4.0",
+				},
+			},
+			configMaps: []*corev1.ConfigMap{
+				newTestConfigMap("eggs2-cluster-values", "org-eggs2"),
+			},
+			secrets: []*corev1.Secret{
+				newTestSecret("eggs2-kubeconfig", "org-eggs2"),
+			},
+			operation: admissionv1.Create,
+			expectedPatches: []mutator.PatchOperation{
+				mutator.PatchAdd("/metadata/annotations", map[string]string{}),
+				mutator.PatchAdd(fmt.Sprintf("/metadata/labels/%s", replaceToEscape(label.AppKubernetesName)), "kiam"),
+				mutator.PatchAdd("/spec/config", map[string]string{}),
+				mutator.PatchAdd("/spec/config/configMap", map[string]string{
+					"namespace": "org-eggs2",
+					"name":      "eggs2-cluster-values",
+				}),
+				mutator.PatchAdd("/spec/kubeConfig/context", map[string]string{
+					"name": "eggs2",
+				}),
+				mutator.PatchAdd("/spec/kubeConfig/secret", map[string]string{
+					"namespace": "org-eggs2",
+					"name":      "eggs2-kubeconfig",
+				}),
+			},
+		},
+		{
+			name:   "case 9: missing cluster label",
```

Comment:
It remains compatible with the current logic. If no `giantswarm.io/cluster` label is set, the correct names for the context and `kubeconfig` secret cannot be obtained, and thus if the validation is run after the mutation, it should fail on missing label.

Reply:
Nice to have a case for this. Can you add a comment saying this will be rejected by the validation webhook?

Reply:
Sure, will add that

Reply:
It's there, also added another case for validation, see this https://github.com/giantswarm/app/pull/208/files#r792571800